src/wp-includes/post.php

```diff
@@ -6499,6 +6514,22 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 		}
 	}
 
+	// Check for alternate full size mime types in the root sources array to delete.
+	if ( isset( $meta['sources'] ) && is_array( $meta['sources'] ) ) {
+		$sources = $meta['sources'];
+		array_shift( $sources );
```

Comment:
I guess the question is, can the original image maybe not be the "primary" type, for example if the original image upload is JPEG and the mapping is to output => array( 'image/webp', 'image/jpeg' ) (webp first), I would expect WebP to always be "primary".

By primary, I mean "the mime type stored in the traditional sizes array" so perhaps looking there instead of the sources array would make more sense.

Reply:
This assumption is `true for new installations but not so much for existing installations, additionally from the performance plugin this only changes the order of the sources not the order of the base image.

One last question is the fact that the base mime type is not part of the `sources` is this expected or a side effect?

Reply:
we may want to adjust that in the performance plugin (or here once we align on approach), did we discuss this already in development of the plugin? What about the case when only one mime type is output (eg WebP), in that case the primary type is not the same as the uploaded file (and no replacements would be needed for WebP type)

> One last question is the fact that the base mime type is not part of the sources is this expected or a side effect?

This was a bug introduced in the last round of changes and should be fixed now. Should be covered in the tests as well which I am working on getting to pass now.

Reply:
maybe something to consider storing, eg a meta value at the root level indicating the primary mime type for that image. Currently no mime type data is stored at the root level.

Reply:
@adamsilverstein @mitogh The "primary" MIME is defined as follows:
* If the original MIME type is included in the list of MIME types to generate (via the filter), it is also the primary MIME.
* If it is _not_ included in the list, the first MIME in the list becomes the primary MIME.
* To know the primary MIME type on an image that already exists (e.g. upload fully complete, with all sizes, sources etc.), we need to look at the image metadata. Whatever MIME type is used in the root of the sub-sized images is the primary MIME. We could of course also run the filter again, but that would be a bit wasteful, and less reliable since the filter values might have changed since the image was uploaded.

In other words, during upload the filter defines the primary MIME type. Once uploaded, the image metadata defines the primary MIME type.

Reply:
> This was a bug introduced in the last round of changes and should be fixed now. Should be covered in the tests as well which I am working on getting to pass now. When you get a chance, please test again with the latest code.

Thanks just reviewing and things are working as expected now, great work with the patch.

> maybe something to consider storing, eg a meta value at the root level indicating the primary mime type for that image. Currently, no mime type data is stored at the root level.

Agree, I think this is determined and stored at the `wp_posts` table instead and can be consumed with `get_post_mime_type`, I would agree that makes more sense and for consistency reasons that storing it here in the meta makes more sense.

> The "primary" MIME is defined as follows:

Agree I think we should amplify the logic to cover a scenario where the default mime type is not placed on the first item on the array instead we should loop normally in all mime types and avoid processing the case where the requested mime type is the default mime type of the uploaded image on the attached file.

Reply:
Thanks for the additional feedback...

> To know the primary MIME type on an image that already exists (e.g. upload fully complete, with all sizes, sources etc.), we need to look at the image metadata. Whatever MIME type is used in the root of the sub-sized images is the primary MIME. 

Right! I'll add storing the Primary mime type at the root level (where we currently don't store a mime type) during image generation (where we already have this primary/secondary logic as described in place), then use this as the primary time on output as well.

Reply:
Added storage in [`253a0df` (#2393)](https://github.com/WordPress/wordpress-develop/pull/2393/commits/253a0dff3c5da6d1b613daa237f8e258c3bdb8f0) update use here in [`bcb13a8` (#2393)](https://github.com/WordPress/wordpress-develop/pull/2393/commits/bcb13a8dad6ed5ccf7e052496a13fb722f9f0373)